plugin/config_test.go

```diff
@@ -101,6 +101,143 @@ output: stdout
 	require.Equal(t, "noop", noop.OperatorType)
 }
 
+type PluginIDTestCase struct {
+	Name          string
+	PluginConfig  pipeline.Config
+	ExpectedOpIDs []string
+}
+
+func TestPluginIDs(t *testing.T) {
+	// TODO: ids shouldn't need to be specified once autogen IDs are implemented
+	pluginContent := []byte(`
+parameters:
+pipeline:
+  - id: noop
+    type: noop
+  - id: noop1
+    type: noop
+`)
+	pluginName := "my_plugin"
+	pluginVar, err := NewPlugin(pluginName, pluginContent)
+	require.NoError(t, err)
+	operator.RegisterPlugin(pluginVar.ID, pluginVar.NewBuilder)
+
+	// TODO: remove ID assignment
+	pluginContent2 := []byte(`
+parameters:
+pipeline:
+  - id: noop3
+    type: noop
+  - id: noop4
```

Comment:
I know we're going to make these unnecessary with #105, but for now, these tests would be stronger if these id's were also `noop` and `noop1`. This would demonstrate that the scoping functionality is differentiating successfully. Also, even with autogenerated IDs, I think these will end up at `noop` and `noop1`.